Socket.io Integration

gint0kix opened the pull request:
Check It Out -Albert Lee 2018

RajendraBhagroo committed:
Initial commit

Commit:
Added .gitignore

RajendraBhagroo committed:
Merge pull request #1 from RajendraBhagroo/Server-Setup

Added .gitignore

Commit:
CircleCI Setup

Commit:
Setup Client, Config, Routes & Package.json

Commit:
Reverted Test Command

Commit:
CircleCI Config Points To Client

RajendraBhagroo committed:
Merge branch 'master' into Server-Setup

Commit:
CircleCI Badge Added To README. See Github Shields

RajendraBhagroo committed:
Update issue templates

RajendraBhagroo committed:
Create CODE_OF_CONDUCT.md

RajendraBhagroo committed:
Merge pull request #2 from RajendraBhagroo/Server-Setup

Server setup

RajendraBhagroo committed:
Merge branch 'master' into Code-Of-Conduct

RajendraBhagroo committed:
Merge pull request #4 from RajendraBhagroo/Code-Of-Conduct

Created CODE_OF_CONDUCT.md

RajendraBhagroo committed:
Merge branch 'master' into Issue-Templates

RajendraBhagroo committed:
Merge pull request #3 from RajendraBhagroo/Issue-Templates

Update issue templates

Commit:
Updated README. Basic Express Server Running. New NPM Scripts

RajendraBhagroo committed:
Merge branch 'master' into Server-Setup

RajendraBhagroo committed:
Merge pull request #5 from RajendraBhagroo/Server-Setup

Updated README. Basic Express Server Running. New NPM Scripts

Commit:
Refactored Server Code Into Its Own Folder For Clarity. Setup Scripts. Updated README. Signed.

Signed-off-by: Rajendra Bhagroo <RajendraBhagroo@Gmail.com>

RajendraBhagroo committed:
Merge branch 'master' into Server-Setup

RajendraBhagroo committed:
Updated README To Be Consistent And Include Dev Servers. Created Wiki!

RajendraBhagroo committed:
Merge branch 'Server-Setup' of https://github.com/RajendraBhagroo/TeleQuest into Server-Setup

RajendraBhagroo committed:
Config Refactored, Reflected In server.js And Updated In README

RajendraBhagroo committed:
Docker Works! Included Docker In README And Created Production Install Script. Docker-Compose Is Next

RajendraBhagroo committed:
Server Setup Complete. User/Profile Models/Routes Basic Implementation.

RajendraBhagroo committed:
Passport.js & JWT Integrated. User Registration Functionality Implemented In API.

RajendraBhagroo committed:
Updated Config Section In README To Include jwt_secret

RajendraBhagroo committed:
Merge pull request #7 from RajendraBhagroo/Api-User-Registration

Api user registration

RajendraBhagroo committed:
Unit Testing Integrated, Common HTTP Methods [GET,POST,DELETE] Included For [Register, Login, Current, Delete] Routes. JWT Bearer Token Works

RajendraBhagroo committed:
Refactored Client/ For Future Work Involving Redux & Unit Testing

RajendraBhagroo committed:
Merge pull request #8 from RajendraBhagroo/Api-Users-HTTP_Methods_And_Testing

Unit Testing Integrated, Common HTTP Methods [GET,POST,DELETE] Includ…

RajendraBhagroo committed:
Merge pull request #9 from RajendraBhagroo/Client-Initial-Setup

Client initial setup

RajendraBhagroo committed:
License change to MIT within package.json & README

RajendraBhagroo committed:
Change License

License change from AGPLv3 to MIT

RajendraBhagroo committed:
MIT License upload

RajendraBhagroo committed:
Removed .md for MIT license to provoke github to change description

RajendraBhagroo committed:
Merge pull request #10 from RajendraBhagroo/License-Change-AGPLv3-MIT

License Change AGPLv3 To MIT

RajendraBhagroo committed:
Refactored configuration file into environment variables

RajendraBhagroo committed:
Updated README To Reflect Migration From Config.js To Environment Variables

RajendraBhagroo committed:
Updated README To Include Code Of Conduct Link In Contributing

RajendraBhagroo committed:
Docker-Compose Works! Nodemon & React live reload successgit add -A Not Optimal For Production

RajendraBhagroo committed:
Refactored Docker configurations for [dev/prod]. Updated Docker README Instructions

RajendraBhagroo committed:
Merge pull request #11 from RajendraBhagroo/Container-Orchestration

Container orchestration

RajendraBhagroo committed:
Seperated Host protocol within server.js, reflect changes in .env

RajendraBhagroo committed:
Updated README To Fix Broken Links When Merged

RajendraBhagroo committed:
React Client Layout [apis,components,redux,resources,history] Constructed. React Developer Tools Integrated. Redux running. This Is A HUGE Commit, Please See App.js & Component Folder Architecture

RajendraBhagroo committed:
Bootstrap & Fontawesome CDN's Included In Client

RajendraBhagroo committed:
Merge pull request #12 from RajendraBhagroo/Client-Redux-Setup

Client redux setup

gint0kix committed:
Finished API unit testing

gint0kix committed:
Resolved Comments

RajendraBhagroo committed:
Merge pull request #13 from RajendraBhagroo/Server-Testing-Users

Implemented API Unit Testing For Users Route

RajendraBhagroo committed:
Updated Server Testing To Sync With Route Versioning V1

RajendraBhagroo committed:
New CricleCI Testing Commands

RajendraBhagroo committed:
New CricleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
New CircleCI Testing Commands

RajendraBhagroo committed:
Server Tests Now Pass Properly

RajendraBhagroo committed:
Refactored scripts to reflect [prod/dev] environments

RajendraBhagroo committed:
Added Timeout For CircleCI

RajendraBhagroo committed:
Adjusted Timeout For CircleCI

RajendraBhagroo committed:
Adjusted Timeout For CircleCI

RajendraBhagroo committed:
CircleCi Test Passes, However Watch Mode Prevents It From Ending

RajendraBhagroo committed:
Added authReducer To Client

RajendraBhagroo committed:
Concurrently Cuasing SIGINT Fail When It Should Not

RajendraBhagroo committed:
Concurrently Cuasing SIGINT Fail When It Should Not

RajendraBhagroo committed:
Merge pull request #14 from RajendraBhagroo/Server-CircleCI-TDD

Server-CircleCI-TDD

RajendraBhagroo committed:
Refactored .env To Work With Client And Refactored Version Numbers For Server Routes/Testing For Re-Usability

gint0kix committed:
Created Profile Model
The current properties of it are
-ClassesTeaching
-Teacher status
-ClassesIn
-School affiliated with
-Public name

RajendraBhagroo committed:
Navbar, Footer Date, About section Implemented. Basic Routing Functional

RajendraBhagroo committed:
Merge pull request #16 from RajendraBhagroo/Client-Authentication

Client-Landing-Setup

RajendraBhagroo committed:
Added Basic Validation For Users Route

RajendraBhagroo committed:
Corrected Users Unit Testing On Server!

RajendraBhagroo committed:
Clever Way To Combat NoSQL Injection Through String Coercion & Refactored Validation Checking Of Users Login/Register. Data Is Validated, Centralized & Sanitized

RajendraBhagroo committed:
Split name field into firstName and lastName. This creates atomic & predicatable values ... and also satisfies unit tests for users route!

RajendraBhagroo committed:
Merge pull request #17 from RajendraBhagroo/Server-Validation

Server-Users-Validation

Gold-Turtle committed:
Added login/register form on the page, and added couse and mypage.

gint0kix committed:
Created profile routes for creation
deletion, adding classes for students, and
adding clases for teachers

Gold-Turtle committed:
Added .env.example on same branch

RajendraBhagroo committed:
Merge pull request #18 from RajendraBhagroo/Authentication-login/register

Added login/register form on the page, and added course and mypage.

RajendraBhagroo committed:
Redux Works For Registration. Updated .env & README. CORS Added for server. CRA was not allowing me to use .env or proxy due to folder structure.

RajendraBhagroo committed:
Removed CORS Whitelist, Was Causing Server Unit Tests To Incorrectly Fail

RajendraBhagroo committed:
Login Works With Redux, No Logout!

RajendraBhagroo committed:
Conditional Navbar Works! Logout Button Implemented

gint0kix committed:
Finished creating routes for profile
Routes include those for
-Student dropping class
-Teacher dropping class they're teaching
-Updating teacher status

Removed the following route
-("/current")
Updated the following route
-("/") now displays the current logged in profile

gint0kix committed:
Revised documentation, the current version
of the update routes adds/drops one class
at a time

RajendraBhagroo committed:
Merge pull request #19 from RajendraBhagroo/Client-Authentication-Redux

Client-Authentication-Redux

RajendraBhagroo committed:
Merge branch 'master' into Server-Profile-Model-Route

RajendraBhagroo committed:
Merge branch 'master' into Server-Profile-Model-Route

gint0kix committed:
lets try this agian

gint0kix committed:
fixed the error

RajendraBhagroo committed:
Merge pull request #21 from RajendraBhagroo/Server-Profile-Model-Route

Server-Profile-Implementation

RajendraBhagroo committed:
Private Routes Implemented. User Model Now Includes userName

RajendraBhagroo committed:
Merge pull request #22 from RajendraBhagroo/Protected-Routes

Private Routes Implemented. User Model Now Includes userName

RajendraBhagroo committed:
Major Refactor. Instead of replicating [v1] everywhere I realized that developers can update the server in source control, but the moment they edit/add a model, they will need to edit/add functionality to routes,tests & validation, therefore breaking existing functionality and moving to a new version. This does not really affect us on the small scale, but as an engineer you should ALWAYS think ahead and what appropriate for the software development lifecycle & scope.

RajendraBhagroo committed:
Updated Profile Model & Routes, Still Need To Add Functionality To Insert/Delete studentFields & teacherFields. Along With Profile Testing & Better Validation. It Is Only Basic Validation Right Now

RajendraBhagroo committed:
Profile Routes Functional. I Tested Them With Postman However Formal Unit Tests Need To Be Written. There Are Glaring Problems In Terms Of Validation, For Now This Is Fine. We Will Need To Remove Restrictions From Models & Enforce Restrictions In Validation Files Soon. After Unit Tests Are Created.

RajendraBhagroo committed:
Merge pull request #23 from RajendraBhagroo/Server-Profile-Update

Code Refactor - Profile Model

RajendraBhagroo committed:
Profile Test Routes Completed. Next Steps: Remove Requirements From Profile Model And Enforce Them In Validation Files. All While Still Passing Tests [Update Profile Tests To Reflect Cases In Profile Validation]

RajendraBhagroo committed:
Merge pull request #24 from RajendraBhagroo/Server-Profile-Testing

Profile Test Routes Completed

RajendraBhagroo committed:
Validation Updated For Profile Creation. I Tested On Postman, Works As Intended. If Student ID Exists, TeacherFields Is Wiped & Vice Versa. This Enforces That You Can Either Be A Student Or Teacher, Not Both!

RajendraBhagroo committed:
Education & Experience Validation For Profile Defined. This Is Not The Best Validation [I Do No Cover Every Case Possible] But It Is A Good Start

RajendraBhagroo committed:
Profile Route Testing Complete [For Now]. I Thought Of The Most Important Test Cases However I May Have Missed Minor Cases And Thats Ok. We Can Get To That Later, I Will Mark It In The Deployment Category Under The Main Project Board. For Time Contraints, This Is Sufficiant.

RajendraBhagroo committed:
Merge pull request #25 from RajendraBhagroo/Server-Profile-Validation

Server-Profile-Validation

gint0kix committed:
Added
-socket.io server-side
-socket.io-streaming
Currently functionallity includes
-Joining specified route and room in route
-Creating specified route, and room in route
-Disconnecting from connection

gint0kix committed:
removed socket.io from root dependancies

gint0kix committed:
Merge branch 'master' into Server-Integrate-SocketIo

RajendraBhagroo commented:
Chris in server.js you used server before defining it. It's causing circle-ci to fail

gint0kix committed:
fixed error

RajendraBhagroo commented:
Chris, chance to root directory and run [npm run circle-ci]. It says socket is not a function.

Gold-Turtle committed:
Profile update form on new file name [ProfileUpdateTest] in home folder

Gold-Turtle committed:
Added Educaton and Experience

Gold-Turtle committed:
Added Public Profile page and profileAction/Reducer, changed Update Profile

RajendraBhagroo committed:
Merge branch 'master' into ProfilePage-UpdateForm

Gold-Turtle committed:
Send profile/experience/education datas to database

Gold-Turtle committed:
Merge branch 'master' of https://github.com/RajendraBhagroo/TeleQuest into ProfilePage-UpdateForm

Gold-Turtle committed:
Merge branch 'ProfilePage-UpdateForm' of https://github.com/RajendraBhagroo/TeleQuest into ProfilePage-UpdateForm

RajendraBhagroo committed:
Tweaked Update Profile Route. We Will Need To Check For Existing Profile Info And Populate Fields If Applicable

RajendraBhagroo committed:
AddExperience Route Reviewed

RajendraBhagroo committed:
AddEducation Reviewed.

RajendraBhagroo committed:
Get Profile Action Implemented. Kim Can Now Make Profile Info Dynamic From Database As opposed To Static. See [profile] Variable!

gint0kix committed:
Finished implementing socket.io server side
Socket.io server will handle the events such as
-connection, when a user connects to the namespace
-joinRoom, when a user joins a specific room on the namespace
-disconnet, when the user leaves a room

gint0kix committed:
Hopefully fixed the error

RajendraBhagroo commented on server/package.json:
```diff
@@ -17,6 +17,8 @@
     "passport": "^0.4.0",
     "passport-jwt": "^4.0.0",
+    "socket.io": "^2.2.0",
+    "socket.io-stream": "^0.9.1",
```
Is this dependency for the server or for the client?

gint0kix replied:
for the server

RajendraBhagroo commented on server/server.js:
```diff
@@ -29,5 +31,44 @@ app.use(`/api/${process.env.SERVER_VERSION}/profile`, profile);
 // Run Server
 const host = process.env.HOST || `127.0.0.1`;
-const port = process.env.NODE_PORT || 3001;
+const port = process.env.NODE_PORT || 3009;
```
Lets try to keep NODE_PORT at 3001 in the .env file and in the server. Try changing the port Socket.io listens on to 3002

RajendraBhagroo commented on server/server.js:
```diff
@@ -29,5 +31,44 @@ app.use(`/api/${process.env.SERVER_VERSION}/profile`, profile);
+    })
+
+
+;
```
Have you tested these changes to ensure that it works properly? Also, try moving this code into its own file and including it in here. Like I did with profile and users, if possible

gint0kix replied:
yeah it works, don't believe its possible, all tutorials have it running on the server js

Gold-Turtle committed:
Added loading but does not work yet, it does not get data

RajendraBhagroo committed:
Profile Spinner & Loading State Corrected

Gold-Turtle committed:
Raj and I finshied profile education/experience add and delete.

Gold-Turtle committed:
Raj and I changed coursesteaching route and tests

Gold-Turtle committed:
Raj and I finished profile routes, onto streams

RajendraBhagroo committed:
Merge pull request #27 from RajendraBhagroo/ProfilePage-UpdateForm

Profile Routes

gint0kix committed:
added new env variable SOCKET_PORT which
represents the port where socket.io will
run on

RajendraBhagroo commented:
Chris, if you added a new variable to our private [.env] file, please add it to the public [.env.example] and update the corresponding code in the repositories readme.md so other developers know to add this variable to their private [.env] files.

Thank you! I will add the variable on circleCI so hopefully your tests stop failing, though they shouldn't fail if you kept express on the same port and move socket.io on another port

gint0kix committed:
Merge branch 'master' into Server-Integrate-SocketIo

Gold-Turtle committed:
Raj and I fixed new account profile bug

RajendraBhagroo committed:
Skills Is Now Treated As A String As Opposed To CSV. We Can Split Into Array When Needed.

RajendraBhagroo committed:
Minor Logic Changes, Code Consistency Changes.

RajendraBhagroo committed:
Merge branch 'master' into ProfilePage-UpdateForm

RajendraBhagroo committed:
Merge pull request #28 from RajendraBhagroo/ProfilePage-UpdateForm

Profile Bug Fixes

gint0kix committed:
Added streaming tab, but no functionality has been
done yet, as well as updated the .env example

gint0kix committed:
merged local and master

gint0kix committed:
Editted stream to App.js to including Streams
Currently impliemented upon load the page will
request ANY user to access their webcam and mic

RajendraBhagroo commented on client/src/components/layout/Navbar.js:
```diff
@@ -28,6 +28,12 @@ class Navbar extends React.Component {
+        <li className="nav-item">
+          <Link to="/streams" className="nav-link">
+            Streams
+            <i className="fas fa-sign-out-alt ml-2" />
```
Remember To Change Icon From Sign Out To Something That Resembles Streams. We Use FontAwsome For Icons, Check Out Their Website https://fontawesome.com/

RajendraBhagroo commented on client/src/components/stream/Stream.js:
```diff
@@ -0,0 +1,34 @@
+import PropTypes from "prop-types";
+
+class Stream extends Component{
+    /*Upon mounting of stream, the user's video, and audio will be obtained
```
See Other Components To Declare State. You Set State On Line 23 For [videoSrc], However Compononet Level State Is Not Defined And [videoSrc] Does Not Exist.

RajendraBhagroo commented on server/server.js:
```diff
@@ -2,10 +2,11 @@ const express = require("express");
 const app = express();
 require(`./${process.env.SERVER_VERSION}/passport`)(passport);
-
+var server = require('http').Server(app);
```
Change var To let

RajendraBhagroo commented on server/server.js:
```diff
@@ -2,10 +2,11 @@ const express = require("express");
 require(`./${process.env.SERVER_VERSION}/passport`)(passport);
-
+var server = require('http').Server(app);
+var io = require('socket.io').listen(server)
```
Change [var] to [let]

RajendraBhagroo committed:
Navbar Logo Now Reflects Streams. Courses Added Are Now Consistent When Displayed [UpperCased]. Streams.js No Longer Throws Exception & Breaks App

RajendraBhagroo committed:
Removed [Course] Page, Changed Default Page To Landing Page

gint0kix committed:
Installed socket.io client side
created two functions one for students
the other for professors. Both will connect to
the main url, which will just be a middle ground
where the professors can display that their class is up

The first function creating the room, with a unique identifier
that was derevied from the class tab that the professor clicked

The second function joining the room said professor created for the
student.

gint0kix committed:
tweaked server.js to better handle rooms, as well to handle streaming
and broadcasting data to the clients in the room
Additionally added socket listeners to react to events
emitted by the server

gint0kix committed:
resolved changed var to let

gint0kix committed:
Video capture works now

RajendraBhagroo committed:
Reviewed Code For Streams

RajendraBhagroo committed:
Added Video Controls

gint0kix committed:
Server should respond to client requests, rtcconnection needs to be implemented first. Will work on tonight

gint0kix committed:
added video recording

gint0kix committed:
Added RTCPeerConnection to handle video streaming, i still need to work on it. Help is needed

gint0kix committed:
added logs for debugging purposes

RajendraBhagroo commented:
@gint0kix  Nice Work!

I was looking over the code and I think we should try to meet before Operating Systems Security to discuss the new code. The Stream start seems to respond, As well as stream end. Video start and end are good. Recording start and stop throw errors for media player. There are also many unused variables.

![image](https://user-images.githubusercontent.com/18294827/57027655-fe187800-6c0a-11e9-861f-37216528c1db.png)

![image](https://user-images.githubusercontent.com/18294827/57027704-2011fa80-6c0b-11e9-89f7-acf249c07b71.png)

gint0kix committed:
streaming works!

Gold-Turtle committed:
Simple frame on stream page

RajendraBhagroo committed:
Streampage is now a 'used' var. Although this component should be refactored [along with man yother things such as no profile page]. This will have to do given our time constraints.

RajendraBhagroo committed:
Converted All Functions To Arrow Functions For Consistency. Tested

RajendraBhagroo committed:
Refactored Stream Functions Into Seperate File

RajendraBhagroo committed:
Refactored ProfileNotFound Code. Moved Stream Functions Into Utils Since It Is Not a Component

RajendraBhagroo committed:
From MDN MediaRecorder, VOD's Work!

RajendraBhagroo committed:
Set Bookmark Button Included, No Functionality Yet

gint0kix committed:
modified stopStream, to clear videoElement, and incoming video

gint0kix committed:
Merge branch 'Server-Integrate-SocketIo' of https://github.com/RajendraBhagroo/TeleQuest into Server-Integrate-SocketIo

Gold-Turtle committed:
Seperate Student View and Professor View, change frames

RajendraBhagroo committed:
Adjusted Video Size, Buttons & Added Simple Bookmark Feature For VOD's

gint0kix committed:
modified webrtc connection, was still set to work on local only

gint0kix committed:
working demo

gint0kix committed:
disabled testing for now

gint0kix committed:
disabled testing for now

Gold-Turtle reviewed (approved):
Approved

RajendraBhagroo commented:
The ICE Issue with RTC Peer Connection Is A False Positive. ICE does not handle stream, Googles STUN servers will. Accepted !

gint0kix committed:
Merge branch 'Server-Integrate-SocketIo' of https://github.com/RajendraBhagroo/TeleQuest into Server-Integrate-SocketIo
